src/pymor/models/iosys.py

```diff
@@ -505,6 +575,44 @@ def to_abcde_files(self, files_basename):
         if E is not None:
             _mmwrite(Path(files_basename + '.E'), E)
 
+    def _compute(self, solution=False, output=False, solution_d_mu=False, output_d_mu=False,
+                 solution_error_estimate=False, output_error_estimate=False,
+                 output_d_mu_return_array=False, mu=None, **kwargs):
+        if not solution and not output:
+            return {}
+
+        # solution computation
+        mu = mu.with_(t=0.)
+        X0 = self.initial_data.as_range_array(mu)
+        input_op = NumpyGenericOperator(
+            lambda U, mu: U * mu['input'],
+            dim_range=self.dim_input,
+            parameters={'input': self.dim_input},
+            linear=True,
+        )
+        rhs = self.B @ input_op
```

Comment:
I wonder if we should also have an `LinearInputOperator` or so? `as_vector` will be called on this `ConcatenationOperator` for every time step, so this might be too inefficient. Maybe we should profile it.

Reply:
Ah, ok, the first approach doesn't work (the function needs to return a NumPy array).

With a specialized `Operator`, I get speedup of about 3.5 for `as_vector`.

```python
import timeit

import numpy as np
from pymor.operators.interface import Operator
from pymor.operators.numpy import NumpyGenericOperator, NumpyMatrixOperator
from pymor.parameters.base import Parameters
from pymor.vectorarrays.numpy import NumpyVectorSpace

n = 100
m = 2

rng = np.random.default_rng(0)
B = NumpyMatrixOperator(rng.standard_normal((n, m)))

input_op = NumpyGenericOperator(
    lambda U, mu: U * mu['input'],
    dim_range=m,
    parameters={'input': m},
    linear=True,
)
rhs = B @ input_op


class InputOperator(Operator):
    linear = True

    def __init__(self, B):
        self.B = B
        self.source = NumpyVectorSpace(1)

    def apply(self, mu=None):
        raise NotImplementedError

    def as_vector(self, mu=None):
        return self.B.as_range_array(mu).lincomb(mu['input'])


Bop = InputOperator(B)

mu = Parameters(input=m).parse('[t, t**2]')
mu = mu.with_(t=2)
num = 10_000
rhs_time = timeit.timeit('rhs.as_vector(mu)', number=num, globals=locals()) / num
Bop_time = timeit.timeit('Bop.as_vector(mu)', number=num, globals=locals()) / num
print('rhs:', rhs_time)
print('Bop:', Bop_time)
print('speedup:', rhs_time / Bop_time)
```

Reply:
Yes, I'd say that `InputOperator` would be the way to go. I would probably implement `as_range_array` instead of `as_vector` to make sure that the impl is called in that case, too. `apply` could be implemented approximately as
```python
B.lincomb(U.to_numpy() * mu['input'][np.newaxis, :])
```

Reply:
I'd also call it `LinearInputOperator`?